### PR TITLE
test(electron): improve coverage from 68% to 80%+ across all metrics

### PR DESCRIPTION
**Pull request by HemSoft**
Add comprehensive tests for 7 electron modules:
- tempoClient: worklog CRUD, schedules, capex map, project links (18 tests)
- ralphService: config, buildArgs, template scripts, process lifecycle (12 tests)
- crewService: addProjectFromPicker, undoFile edge cases (7 tests)
- dispatcher: skill worker, failure handling, snapshot collection (5 tests)
- terminalHandlers: PTY data/exit, OSC 7 CWD, spawn failure (5 tests)
- shellHandlers: fetch-page-title DNS/redirect/non-HTML paths (6 tests)
- telemetry: initTelemetry, recordIpcCall, withSpan edge cases (3 tests)

Coverage improvements (electron/):
  Statements: 67% → 80% (+13pp)
  Branches:   52% → 68% (+16pp)
  Functions:  69% → 79% (+10pp)
  Lines:      69% → 82% (+13pp)

Ratchet coverage thresholds to lock in gains.

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: vitest.electron.config.ts [hunk 1]
```diff
@@ -17,10 +17,10 @@ export default defineConfig({
...
       ],
       thresholds: {
-        statements: 67,
-        branches: 52,
-        functions: 69,
-        lines: 69,
+        statements: 80,
+        branches: 67,
+        functions: 78,
+        lines: 81,
```
[the remark this answers is not in the document]
Comment (HemSoft) [reply vitest.electron.config.ts]: The current thresholds (statements: 80, branches: 67, functions: 78, lines: 81) are already within ~1% of the achieved coverage (80.37/67.79/78.78/81.85). Setting them to the exact achieved values (e.g., 80.37) would cause flaky CI failures from minor coverage fluctuations. The current integer floor values are appropriate — they prevent meaningful regression while allowing normal variance.

File: electron/ipc/terminalHandlers.test.ts [hunk 2]
```diff
@@ -218,10 +256,110 @@ describe('terminalHandlers', () => {
...
+
+    expect(ptyState.callbacks.data).toBeTypeOf('function')
+    ptyState.callbacks.data!('\u001b]7;file:///C:/workspace/repo\u0007')
+
+    expect(mockSender.send).toHaveBeenCalledWith(
+      IPC_PUSH.TERMINAL_CWD_CHANGED,
+      spawnResult.sessionId,
+      'C:\\workspace\\repo'
+    )
+  })
```
[the remark this answers is not in the document]
Comment (HemSoft) [reply electron/ipc/terminalHandlers.test.ts]: Fixed. Changed the mock to use a Unix path (`/workspace/repo`) and the assertion to use `resolve('/workspace/repo')` so it's platform-correct on both Windows and Linux. The test now passes on all platforms.

File: electron/telemetry.test.ts [hunk 3]
```diff
@@ -158,4 +158,36 @@ describe('telemetry', () => {
       await expect(shutdownTelemetry()).resolves.toBeUndefined()
     })
   })
+
+  describe('initTelemetry', () => {
+    it('is a no-op when OTEL_EXPORTER_OTLP_ENDPOINT is not set', async () => {
+      const { initTelemetry } = await import('./telemetry')
+      // process.env has no OTEL_EXPORTER_OTLP_ENDPOINT by default in test
+      await expect(initTelemetry()).resolves.toBeUndefined()
+    })
```
[the remark this answers is not in the document]
Comment (HemSoft) [reply electron/telemetry.test.ts:770]: Fixed. Added save/restore of `OTEL_EXPORTER_OTLP_ENDPOINT` env var with try/finally block to make the test deterministic regardless of CI environment.

File: electron/services/crewService.test.ts [hunk 4]
```diff
@@ -1,9 +1,18 @@
 import { describe, it, expect, vi, beforeEach } from 'vitest'
 
+const { mockShowOpenDialog, mockExecFileAsync, mockParseGitRemote, mockIsGitHubHost } = vi.hoisted(
+  () => ({
+    mockShowOpenDialog: vi.fn().mockResolvedValue({ canceled: true, filePaths: [] }),
+    mockExecFileAsync: vi.fn().mockResolvedValue({ stdout: '' }),
+    mockParseGitRemote: vi.fn(() => null),
+    mockIsGitHubHost: vi.fn(() => false),
+  })
+)
```
[the remark this answers is not in the document]
Comment (HemSoft) [reply electron/services/crewService.test.ts]: Fixed. Changed `vi.fn(() => null)` to `vi.fn().mockReturnValue(null)` and `vi.fn(() => false)` to `vi.fn().mockReturnValue(false)`. This creates `Mock<Procedure>` accepting any args, fixing both the TS2556 spread errors (lines 41-42) and the TS2345 return type errors (lines 325, 372).

File: electron/services/ralphService.test.ts [hunk 5]
```diff
@@ -256,4 +278,326 @@ describe('ralphService', () => {
...
+        if (filePath === repoPath) return true
+        if (filePath === `${repoPath}\\scripts\\ralph-review.ps1`) return true
+        return false
+      })
+
+      const result = launchLoop({
+        repoPath,
+        scriptType: 'template',
+        templateScript: 'ralph-review.ps1',
+      } as Parameters<typeof launchLoop>[0])
```
[the remark this answers is not in the document]
Comment (HemSoft) [reply electron/services/ralphService.test.ts]: Fixed. Changed all `'C:\\repo'` to `'/repo'` which is recognized as absolute on all platforms. Also fixed the template script tests to compute `scriptsDir` dynamically via `resolve(dirname('/mock/service'), '..', 'scripts/ralph-loops')` instead of using `filePath.includes('ralph-loops')` which was too broad and matched vendored paths.
